engine/player/unique_gear_dragonflight.cpp

```diff
@@ -1815,6 +1815,18 @@ void spoils_of_neltharus( special_effect_t& effect )
       proc_spell_t::execute();
 
       buff_list.front().first->expire();
+
+      if ( sim->current_time() == timespan_t::zero() )
```

Comment:
You can now check the `is_precombat` bool in the action to see if it's being executed in the precombat apl. However, this entire section is unnecessary. Instead of individual string comparison, you can edit the init_buff lambda in the constructor here https://github.com/simulationcraft/simc/blob/dragonflight/engine/player/unique_gear_dragonflight.cpp#L1784 to `emplace_front` instead if the option string matches `n`. And later in the shuffle line here https://github.com/simulationcraft/simc/blob/dragonflight/engine/player/unique_gear_dragonflight.cpp#L1803 only shuffle is `!is_precombat`

Reply:
I tried doing what you suggested. The `emplace` part works, I even added a `print_debug` to make sure the `buff_list` is in the correct order after the 4 `init_buff` lines. I moved the shuffle to `void init_finished() override`, and then commented it out for testing (the problem of `is_precombat` is that the trinket can't be used in precombat currently as it's flagged as harmful, but that's not the main issue right now).
But the buff obtained is random, as when I use the trinket only on pull each of the 4 stats is obtained 25% of the time, indicating the callback is executed. This is still true when the trinket is used as the first action of the sim, with nothing in precombat. But for some reason, when I turn on `debug=1`, the trinket always give the chosen buff in the log. Is the `buff_list` initialized only in the first iteration and not at the beginning of each?
I I'm pretty lost on what's happening here, and would gladly take any help.

I can commit my non-functional work if it helps resolving the issue. We can also move this discussion on Discord, I'm on the simc server.